src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -135,16 +145,23 @@ public void run() {
                 if (runPolling()) {
                     GitHubPushCause cause;
                     try {
-                        cause = new GitHubPushCause(getLogFile(), pushBy);
+                        File logFile = getLogFile();
+                        if (null == logFile) {
+                            throw new IOException("Log file was null.  Job was probably null.");
+                        }
+
+                        cause = new GitHubPushCause(logFile, pushBy);
                     } catch (IOException e) {
                         LOGGER.warn("Failed to parse the polling log", e);
                         cause = new GitHubPushCause(pushBy);
                     }
-                    if (asParameterizedJobMixIn(job).scheduleBuild(cause)) {
+
+                    if (null != job && asParameterizedJobMixIn(job).scheduleBuild(cause)) {
                         LOGGER.info("SCM changes detected in " + job.getFullName()
                                 + ". Triggering #" + job.getNextBuildNumber());
                     } else {
-                        LOGGER.info("SCM changes detected in " + job.getFullName() + ". Job is already in the queue");
+                        LOGGER.info("SCM changes detected in " + (null == job ? "null" : job.getFullName())
```

Comment:
Presumably if `job == null` you could just return early from `run`?

Reply:
I think the impetus here was the code quality checker complaining about no null check, I'll see if I can add it further up, that is a better way to handle it, and can leave this line unchanged.

Reply:
Yeah, spot bugs yells at me:

```

[2019-03-27T20:06:25.805Z] [ERROR] Possible null pointer dereference of hudson.triggers.Trigger.job in com.cloudbees.jenkins.GitHubPushTrigger$1.run() [com.cloudbees.jenkins.GitHubPushTrigger$1, com.cloudbees.jenkins.GitHubPushTrigger$1] Dereferenced at GitHubPushTrigger.java:[line 167]Known null at GitHubPushTrigger.java:[line 163] NP_NULL_ON_SOME_PATH
```
adding back the null check.